src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -1051,6 +1039,20 @@ private Type EmitExprListNode(AssociativeNode node)
                 Type t = DfsTraverse(el);
                 if (t == null) return;
 
+                //if this element is a CLRStackValue, we need to unmarshal it.
+                if (t == typeof(DSASM.CLRStackValue))
+                {
+                    //reorganize stack.
+                    EmitOpCode(OpCodes.Pop);
+                    EmitOpCode(OpCodes.Ldarg_2);
+                    DfsTraverse(el);
```

Comment:
Can you explain what you're doing here? I'm confused.

Reply:
I added an `unmarshal` instance util function to the `output` dictionary since I was already cacheing a marshaler there -
this code pops the current arg off the stack, so we can put the `output` dictionary on top first, then adds back the arg - thats done because we want to call an instance method on the `output` dictionary.

I did not try it but perhaps I can use the `ASTTypeMap` or variables dictionary instead to avoid the pop and extra stack load.

Reply:
Ok, makes sense. Yes, I was wondering if you could avoid the first `DfsTraverse(el)` in line 1039 but I suppose this is okay for now.

Reply:
So, if I understand this correctly, you need to unmarshal each element in an array in addition to the top-level unmarshal you're doing at the function call level for each argument, correct?

Reply:
so - maybe I am misunderstanding your question - but the reason I am adding any code here at all is a failing test - consider this case:

x = [1,2,3]
y = vaueAtIndex(x,1) // this replicates for some reason. - so `y` is a `CLRStackValue`
z  = [1,y];

z now contains `[int,CLRStackValue]`

We can't really defer this safely because the unmarshal methods always assume that `CLRStackValues` are always nested only under other `CLRStackValues` - so we unmarshal any wrappers we find when creating a list to avoid a situation where a plain c# object list might contain a CLRStackValue element.

Reply:
Cool, thanks!